fideslog/api/main.py

```diff
@@ -28,6 +29,31 @@
 app.include_router(api_router)
 
 
+# Defined before `log_request` and `require_version_header` to ensure that each are always executed.
+@app.middleware("http")
+async def require_access_token(request: Request, call_next: Callable) -> Response:
```

Comment:
This needn't be a middleware, FastAPI has a pattern for this at individual endpoint level e.g. https://github.com/ethyca/fides/blob/0ed1baab371dfb8f5a8077c94d285c1ac82a0102/src/fides/api/ops/api/v1/endpoints/config_endpoints.py#L20

Reply:
Correct - but FastAPI does not support the type of authentication that I've implemented, so this pattern wouldn't work.